packages/webdriver/src/request.js

```diff
@@ -38,9 +38,13 @@ export default class WebDriverRequest extends EventEmitter {
     }
 
     makeRequest (options, sessionId) {
-        const fullRequestOptions = Object.assign({}, this.defaultOptions, this._createOptions(options, sessionId))
+        let fullRequestOptions = Object.assign({}, this.defaultOptions, this._createOptions(options, sessionId))
+        if (typeof options.transformRequest === 'function') {
+            fullRequestOptions = options.transformRequest(fullRequestOptions)
+        }
+
         this.emit('request', fullRequestOptions)
-        return this._request(fullRequestOptions, options.connectionRetryCount)
+        return this._request(fullRequestOptions, options.connectionRetryCount, 0, options.transformResponse)
```

Comment:
Let's change order of arguments here so that we can still work with defaults.

```suggestion
        return this._request(fullRequestOptions, options.transformResponse, options.connectionRetryCount)
```

Reply:
The pattern is usually: required parameters first, optional last